content/blog/2024-09-10-memory-access-amortization.md

```diff
@@ -0,0 +1,158 @@
++++
+# `title` is how your post will be listed and what will appear at the top of the post
+title= "Memory Access Amortization in Valkey "
+# `date` is when your post will be published.
+# For the most part, you can leave this as the day you _started_ the post.
+# The maintainers will update this value before publishing
+# The time is generally irrelvant in how Valkey published, so '01:01:01' is a good placeholder
+date= 2024-09-11 01:01:01
+# 'description' is what is shown as a snippet/summary in various contexts.
+# You can make this the first few lines of the post or (better) a hook for readers.
+# Aim for 2 short sentences.
+description= "Improve performance by enabling the processor's speculative execution capability"
+# 'authors' are the folks who wrote or contributed to the post.
+# Each author corresponds to a biography file (more info later in this document)
+authors= [ "dantouitou", "uriyagelnik"]
++++
+<style>
+  .inverted-code {
+    font-family: "Courier New", Courier, monospace;
+    background-color: #333; /* Dark background */
+    color: #f0f0f0;         /* Light text */
+    padding: 10px;
+    border-radius: 5px;
+    display: block;
+    white-space: pre-wrap;
+  }
+</style>
+
+ In the [first part](https://valkey.io/blog/unlock-one-million-rps/) of this blog, we described how we offloaded almost all I/O operations to I/O threads, thereby freeing more CPU cycles in the main thread to execute commands. When we profiled the execution of the main thread, we found that a considerable amount of time was spent waiting for external memory. This was not entirely surprising, as when accessing random keys, the probability of finding the key in one of the processor caches is relatively low.  Considering that external memory access latency is approximately 50 times higher than L1 cache, it became clear that despite showing 100% CPU utilization, the main process was mostly “waiting”. In this blog, we describe the technique we have been using to increase the number of parallel memory accesses, thereby reducing the impact that external memory latency has on performance.
```

Comment:
You can drop the domain name from the link. E.g. `[first part](/blog/unlock-one-million-rps/)`

This helps with SEO (marginally).